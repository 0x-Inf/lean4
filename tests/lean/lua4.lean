Variable x : Int

(**
-- Add a variable to the environment using Lua
-- The type of the new variable is equal to the type
-- of x
local env = get_environment()
typeofx = env:check_type(Const("x"))
print("type of x is " .. tostring(typeofx))
env:add_var("y", typeofx)
**)

Check x + y
