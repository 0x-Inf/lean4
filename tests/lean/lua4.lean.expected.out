  Set: pp::colors
  Set: pp::unicode
  Imported 'Int'
  Assumed: x
type of x is ℤ
x + y : ℤ
